### PR TITLE
feat(FX-4757, FX-4758): Create new artwork list UI

### PR DESCRIPTION
**Pull request by dimatretyak**
This PR resolves [FX-4757], [FX-4758] 

#### iOS
https://user-images.githubusercontent.com/3513494/234941973-2208aead-eddc-4950-a8bf-2ef9a4acc13e.mp4

#### Android
https://user-images.githubusercontent.com/3513494/235151341-291468fd-0e47-46a1-9960-53be87fde790.mp4

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Create new artwork list UI - dimatretyak

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[FX-4757]: https://artsyproduct.atlassian.net/browse/FX-4757?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[FX-4758]: https://artsyproduct.atlassian.net/browse/FX-4758?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Components/ArtworkLists/views/CreateNewArtworkListView/components/CreateNewArtworkListForm.tsx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+              <Spacer y={4} />
+
+              <Button
+                width="100%"
+                block
+                disabled={isSaveButtonDisabled}
+                loading={formik.isSubmitting}
+                onPress={formik.handleSubmit}
+              >
```
Comment (rajsam003) [inline src/app/Components/ArtworkLists/views/CreateNewArtworkListView/components/CreateNewArtworkListForm.tsx:134]:
```suggestion
              <Button
                block
                disabled={isSaveButtonDisabled}
                loading={formik.isSubmitting}
                onPress={formik.handleSubmit}
              >
```

We dont need width="100%" if it has block prop

File: src/app/Components/ArtworkLists/views/CreateNewArtworkListView/components/CreateNewArtworkListForm.tsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+                Save
+              </Button>
+
+              <Spacer y={1} />
+
+              <Button width="100%" block variant="outline" onPress={closeCurrentView}>
```
Comment (rajsam003) [inline src/app/Components/ArtworkLists/views/CreateNewArtworkListView/components/CreateNewArtworkListForm.tsx:140]: same

File: src/app/Components/ArtworkLists/views/CreateNewArtworkListView/components/CreateNewArtworkListForm.tsx (hunk 1 above)
Comment (dimatretyak) [reply src/app/Components/ArtworkLists/views/CreateNewArtworkListView/components/CreateNewArtworkListForm.tsx:134]: yeah, you're right. good catch!

Comment (dimatretyak) [reply src/app/Components/ArtworkLists/views/CreateNewArtworkListView/components/CreateNewArtworkListForm.tsx]: Addressed here: https://github.com/artsy/eigen/pull/8593/commits/4ed1197ac80eafeda103b05cad9314dbf6e9b6ef

**Review by rajsam003 — DISMISSED** [review]
LGTM

**Pull request merged**